Correct :info flash type styling confusion

rmoorman opened the pull request:
Correct the display of `:info` flash messages to be in line with what
would be expected from an informational message (blue with "Info" in the
title).

Before this change, flash messages of kind `:info` would be green and
have "Success!" in their title. This is wrong, as success-style feedback
should only be shown on clearly successful actions and in cases neutral
feedback is needed, success-style feedback does not fit.

One example for this is the password reset, when we don't want to
disclose whether or not a reset email has been sent. A green "Success!"
flash message doesn't fit there.

The other way around, a more neutral blue flash message does also fit in
situations the user should receive positive feedback.

In cases a distinct `:success` flash message kind is needed, it can be
added in the project's core components.

lifeiscontent commented on installer/templates/phx_web/components/core_components.ex:
```diff
@@ -46,13 +46,15 @@ defmodule <%= @web_namespace %>.CoreComponents do
       class={[
         "fixed top-2 right-2 mr-2 w-80 sm:w-96 z-50 rounded-lg p-3 ring-1",
-        @kind == :info && "bg-emerald-50 text-emerald-800 ring-emerald-500 fill-cyan-900",
+        @kind == :info && "bg-blue-50 text-blue-800 ring-blue-500 fill-blue-900",
```
personally, I would use cyan or sky for info messages, but it may be bias from seeing so many UI Frameworks using the word "info" with a light blue color

rmoorman replied:
I guess sky it will be then :-)

SteffenDE commented:
Hi @rmoorman,
thank you for contributing! Did you see https://github.com/phoenixframework/phoenix/pull/5844 and especially https://github.com/phoenixframework/phoenix/pull/5844#issuecomment-2218478212?

rmoorman commented:
Oh my! @SteffenDE , I indeed didn't see that.

I suppose @josevalim didn't have a change of heart yet in this matter? 😄 

IMHO it is likely that at least one neutral message kind/style is often needed (instead of green).

On one hand this could be added (like in this PR at the moment). In case it would be feasible/wanted to move this forward, of course I would also amend the inline documentation.

But on the other hand, indeed the `:info` style could be adjusted to be more neutral without adding another kind. I can also amend this PR to give the info flash message a more neutral styling.

Just let me know how you would like me to proceed.

glhrmv commented:
Personally I think this introduces a very useful and intuitive option that most people would likely self-implement. An "official" option makes total sense.

SteffenDE commented:
In order to move this forward, I'd say let's just change info to use sky (I visually prefer it to blue):

https://play.tailwindcss.com/Bn7JbDMpq4

<img width="410" alt="image" src="https://github.com/user-attachments/assets/adecfb62-640c-4086-a807-25ffcb2c6b0f">

People can easily extend their flash types themselves. If the docs aren't clear enough, we should improve those (see also https://github.com/phoenixframework/phoenix_live_view/pull/3344) :)

josevalim commented:
@SteffenDE agreed.

rmoorman commented:
I will change the color to sky then.
You also seem to prefer "Info" instead of "Please note", right? I can see that it might be more in line with the single-word flash titles for the error and success cases. Also it is more grep-able and matches the status itself. I will change that as well.
And it seems there might be some conflicts too to resolve here :-)

rmoorman commented:
So I went on and did the following
* resolved a few minor conflicts (phoenixframework/phoenix#5920 and phoenixframework/phoenix#5979)
* adjusted the color of the flash messages to use the `sky` tint instead of `blue`
* adjusted the icon of the success flash message to use a `check` icon instead of an `information` icon
* changed the info flash message title to just read "Info" (instead of "Please note")
* opened up phoenixframework/phoenix_live_view#3534 to adjust the inline documentation to match the changes here

SteffenDE commented:
@rmoorman I think there was a misunderstanding. With my comment I wanted to say that we should keep just info and error for now, but adjust info to be less "successy", i.e., change it from green to blue. Instead of the dark blue my preference was the lighter sky blue. I do see that I didn't express this as clearly as I should have, so now I'm myself not sure if that's also what @josevalim agreed on 😅

josevalim commented:
@SteffenDE I am in agreement with you :) That's what I had in mind too (just change the color).

rmoorman commented:
So, you want the following adjustments @SteffenDE and/or @josevalim:

- remove the `:success` component variant; effect being that only `:info` and `:error` remain, as it was basically, but now with the color changed to "sky", "Info" instead of "Success" in the flash message's title, and that at least the confusion between `:info` kind and "success" title and styling is cleaned up
- adjust the (live view/components) documentation (e.g. [here](https://github.com/phoenixframework/phoenix_live_view/blob/main/lib/phoenix_live_view.ex#L689)) to make it more obvious that one meant to add own states to flash (like success) if needed

Is that correct?

(P.s. I still really find it a pity because (as others mentioned as well) it seems to be a common thing to have)

SteffenDE commented:
@rmoorman yes!

lifeiscontent commented:
> (P.s. I still really find it a pity because (as others mentioned as well) it seems to be a common thing to have)

I think the reasoning here for not adding success is phoenix has already been using `:info` so in the effort of not changing things out from under people, @SteffenDE is suggesting to continue to use `:info` for that reason,

and I agree with this, live view currently doesn't have any strong opinions, if you want to add a success type to flashes, its relatively straight forward, and for all the generated code, you still have the ability to change it, the framework isn't stopping you in any way to use `:success` in your own application.

rmoorman commented:
@SteffenDE , I adjusted the PR to match the requested changes. It now only adjusts the styling and the title of the `:info` flash messages.

josevalim commented on installer/templates/phx_web/components/core_components.ex:
```diff
@@ -77,7 +77,7 @@ defmodule <%= @web_namespace %>.CoreComponents do
     ~H"""
     <div id={@id} aria-live="polite">
-      <.flash kind={:info} title=<%= maybe_heex_attr_gettext.("Success!", @gettext) %> flash={@flash} />
+      <.flash kind={:info} title=<%= maybe_heex_attr_gettext.("Info", @gettext) %> flash={@flash} />
```
Reading "Info" or "Info!" feels weird. Maybe we can just get rid of the title? Let's just use an icon instead?

SteffenDE replied:
![image](https://github.com/user-attachments/assets/bfc11bc6-584d-4657-a051-3d145dcbf8c4)

For reference: that's how it currently looks

josevalim replied:
I would kill the title fully. Just have the message. The color and the icon already tell me enough about it.

SteffenDE replied:
<img width="407" alt="image" src="https://github.com/user-attachments/assets/30616c41-6f75-46e6-841e-8121a0658536" />

Multi-line messages
<img width="408" alt="image" src="https://github.com/user-attachments/assets/ddd7df84-3085-4973-bbe9-d67a492e0d26" />

Info
<img width="416" alt="image" src="https://github.com/user-attachments/assets/0032f6ae-3924-487e-a695-c98dfc140f92" />

I increased the icon size to match the text's line-height, otherwise it would not look aligned:

<img width="402" alt="image" src="https://github.com/user-attachments/assets/2a0b5f14-5b47-4fe2-bad1-6dacfe883915" />

SteffenDE replied:
I opened https://github.com/phoenixframework/phoenix/pull/6065.

SteffenDE commented:
Closing in favor of https://github.com/phoenixframework/phoenix/pull/6065. Thank you very much for starting this @rmoorman!
